### PR TITLE
Fix #306509: Improvements to zoom functionality

### PR DESCRIPTION
**Pull request by Spire42**
Resolves: [#306509](https://musescore.org/en/node/306509)

Made several significant usability improvements to the zoom functionality, and also fixed several bugs.

Although there are several different changes contained here, the code for them is all intertwined and it would have been difficult and time-consuming to attempt to separate them all into separate PRs.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/WorkflowAndGuidelines/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by anatoly-os** [comment]
Please remove that "merge" commit by squashing the commits.

**Comment by MarcSabatella** [comment]
Can you summarize the changes?  I saw some discussion of changes in where the focus is while zooming, and it is important to understand that while for some of these cases the current behavior might be not optimal even indeed likely to be purely accidentals of the implementation, some of these have been carefully crafted over the years in response to extensive input from users, and it is important not to disturb those cases.

Most important I can remember offhand - Ctrl+wheel must continue to center on the mouse pointer, zoom out and zoom in should return to same position where possible.

**Comment by Spire42** [comment]
> Can you summarize the changes?

I wrote about them in great detail in issue [#306509](https://musescore.org/en/node/306509).

Nothing is disturbed other than how much each zoom step zooms in or out, which is now configurable instead of hard-coded. Mouse zoom relative to the pointer is unchanged.

**Comment by MarcSabatella** [comment]
Great, thanks.  Looking forward to the saner steps for keyboard zoom and relationship to mouse!
